### ISSUE TITLE
docker:tag fails with: Cannot invoke "io.fabric8.maven.docker.config.BuildImageConfiguration.cleanupMode()" because "buildConfig" is null

### ISSUE THREAD

**Issue by OndraZizka** [fabric8io/docker-maven-plugin#1610]
### Description

I'm trying to create a docker tagged image with Docker Maven Pugin.

The build goal works, but when trying to tag the image, DMP fails:

`Execution start of goal io.fabric8:docker-maven-plugin:0.40.2:tag failed: Cannot invoke "io.fabric8.maven.docker.config.BuildImageConfiguration.cleanupMode()" because "buildConfig" is null`

Same if I try with `mvn docker:tag`:
` Execution default-cli of goal io.fabric8:docker-maven-plugin:0.40.2:tag failed: Cannot invoke "io.fabric8.maven.docker.config.BuildImageConfiguration.cleanupMode()" because "buildConfig" is null`

### Info

* docker-maven-plugin version :  0.40.2 (latest 2022-09)
* Maven version (`mvn -v`) :
```
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /sw/prog/maven-3.8.6
Java version: 18.0.2-ea, vendor: Private Build, runtime: /usr/lib/jvm/java-18-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-48-generic", arch: "amd64", family: "unix"
```

* Docker version : v20.10.17
* If it's a bug, how to reproduce :  Use the docker:tag goal.
* Sample project : I will add a repo later.

**Comment by OndraZizka** [comment]
I've also posted a StackOverflow question in case it's my fault: https://stackoverflow.com/questions/73933182/fabric8-docker-maven-plugin-dockertag-fails-with-cannot-invoke-buildimageconf

**Comment by rhuss** [comment]
Yes, it could be that a `mvn docker:tag` iterates over all image configurations and fails if a build section is missing (which is the case for your mongo DB image). If so, this is a bug.
Happy to take any help for fixing this.

As alternative you might consider to leverage the `<tags>` option in the build configuration for tagging, instead of docker:tag.
